### PR TITLE
Fix understat functions

### LINKED ISSUE #239: Understat shot functions all broken
All `understat_` shooting functions are no longer returning results. It appears as though the `script` with the `var name` is no longer exposed to `rvest` scraping.

Will investigate a solution.

### PR DESCRIPTION
**Pull request by tonyelhabr**
Fixes #239. It seems that understat implemented something where you must pass a cookie to the page. I believe this cookie should be the same for all people. [I found this SO post that led me to the solution.](https://stackoverflow.com/questions/43754907/php-file-get-contents-doesnt-get-all-contents-try-with-curl-same-result)

### CODE REVIEW

File: R/understat_teams.R [hunk 1]
```diff
@@ -52,55 +52,57 @@ understat_team_stats_breakdown <- function(team_urls) {
 }
 
 .understat_team_stats_breakdown <- function(team_url) {
-  data_html <-
```
Comment (tonyelhabr) [inline R/understat_teams.R:55]: the code diff looks big here. really, it's just one refactor at the top here + the addition of an `if(!is.na(data_html))` check

File: R/internals.R [hunk 2]
```diff
@@ -245,6 +245,13 @@
   return(clean_val)
 }
 
+.get_understat_json <- function(page_url) {
+  tryCatch(
+    httr::GET(page_url, httr::set_cookies(.cookies = c("beget" = "begetok"))) %>% httr::content(),
```
Comment (tonyelhabr) [inline R/internals.R:250]: this is the major change. i think what happens is the page is loaded twice now. the first time, the page loads on the server-side as it used to + adds the cookie requirement, then loads a second time. if the cookie isn't provided by the user on the second request, then the page is not loaded. i'm guessing they implemented this to (try to) prevent scraping

File: man/load_understat_league_shots.Rd [hunk 3]
```diff
@@ -23,7 +23,7 @@ Returns shooting locations for all matches played in the selected league
 \examples{
 \dontrun{
 try({
-df <- load_understat_league_shots(league="Serie A")
+  df <- load_understat_league_shots(league = "Serie A")
```
Comment (tonyelhabr) [inline man/load_understat_league_shots.Rd:26]: this happened by itself when i called `devtools::document()`

**Comment by tonyelhabr** [comment]
There are lots of test failures. AFAICT there is 1 fotmob-related test failure, and the rest are fbref-related. I suppose the lack of understat test failures is good news tho.

File: R/internals.R (hunk 2 above)
Comment (JaseZiv) [reply R/internals.R:250]: Such a good find! Thanks heaps

**Review by JaseZiv — APPROVED** [review]

**Pull request merged**